### PR TITLE
Eliminate try-catch from json_array_contains Presto function

### PR DESCRIPTION
**Pull request by mbasmanova**
Differential Revision: D58088803

### CODE REVIEW

File: velox/functions/prestosql/SIMDJsonFunctions.h [hunk 1]
```diff
@@ -71,52 +71,39 @@ struct SIMDJsonArrayContainsFunction {
...
+                simdjson::ondemand::number_type::floating_point_number &&
+            !v.get_double().get(doubleValue) && doubleValue == value) {
+          result = true;
+          break;
+        }
+      } else {
+        std::string_view stringValue;
+        if (v.type() == simdjson::ondemand::json_type::string &&
+            !v.get_string().get(stringValue)) {
+          if (((std::string_view)value).compare(stringValue) == 0) {
```
Comment (Yuhta) [inline velox/functions/prestosql/SIMDJsonFunctions.h:103]: nit: two `if`s can be merged

File: velox/functions/prestosql/SIMDJsonFunctions.h [hunk 2]
```diff
@@ -71,52 +71,39 @@ struct SIMDJsonArrayContainsFunction {
...
-          }
-        } else if constexpr (std::is_same_v<TInput, double>) {
-          if (v.type() == simdjson::ondemand::json_type::number &&
-              v.get_number_type() ==
-                  simdjson::ondemand::number_type::floating_point_number &&
-              v.get_double() == value) {
+      if constexpr (std::is_same_v<T, bool>) {
+        bool boolValue;
+        if (v.type() == simdjson::ondemand::json_type::boolean &&
+            !v.get_bool().get(boolValue) && boolValue == value) {
```
Comment (Yuhta) [inline velox/functions/prestosql/SIMDJsonFunctions.h:77]: To match the old code, if the error here is not `INCORRECT_TYPE` or `NUMBER_ERROR`, we should return null immediately.  We maybe need to check this behavior to match what Java is doing.

Comment (mbasmanova) [reply velox/functions/prestosql/SIMDJsonFunctions.h:77]: My reading of the Java code suggests that there is no 'early termination' logic. Java returns NULL only on exception, end of document, end of array.

presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonFunctions.java

```
        try (JsonParser parser = createJsonParser(JSON_FACTORY, json)) {
            if (parser.nextToken() != START_ARRAY) {
                return null;
            }

            while (true) {
                JsonToken token = parser.nextToken();
                if (token == null) {
                    return null;
                }
                if (token == END_ARRAY) {
                    return false;
                }
                parser.skipChildren();

                if ((token == VALUE_NUMBER_INT) &&
                        ((parser.getNumberType() == NumberType.INT) || (parser.getNumberType() == NumberType.LONG)) &&
                        (parser.getLongValue() == value)) {
                    return true;
                }
            }
        }
        catch (IOException e) {
            return null;
        }
```

Comment (mbasmanova) [reply velox/functions/prestosql/SIMDJsonFunctions.h:77]: There is a sanity check for 'double' though. This is missing in Velox. I'll make a separate PR to add.

```
        if (!Doubles.isFinite(value)) {
            return false;
        }
```

**Review by Yuhta — APPROVED** [review]

**Pull request closed without merging**